### PR TITLE
Feature 3.7/improved observability

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

This PR improves observability in the following ways:

* Add database, shard name and error information to several shard-related log messages.
* Display shard names of a collection in the web interface when in the details view of the collection.
* Added HTTP requests metrics for tracking the number of superuser and normal user requests separately:
  - `arangodb_http_request_statistics_superuser_requests`: Total number of HTTP requests executed by superuser/JWT
  - `arangodb_http_request_statistics_user_requests`: Total number of HTTP requests executed by clients

This PR also contains all commits from https://github.com/arangodb/arangodb/pull/12753, which will make no difference when this gets merged into 3.7.

- [ ] :hankey: Bugfix 
- [x] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x]:muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12088/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12152/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12156/

**Comment by jsteemann** [comment]
Tests blue

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12163/

**Comment by jsteemann** [comment]
Tests blue

File: arangod/Cluster/SynchronizeShard.cpp [hunk 1]
```diff
@@ -566,7 +566,8 @@ static arangodb::ResultT<SyncerId> replicationSynchronize(
   SyncerId syncerId{syncer->syncerId()};
 
   try {
-    Result r = syncer->run(configuration._incremental);
+    std::string const context = "synchronization of shard " + col->name() + " in database " + database;
+    Result r = syncer->run(configuration._incremental, context.c_str());
```
Comment (dothebart) [inline arangod/Cluster/SynchronizeShard.cpp:570]: do we really want to downmix to a c-string here?

Comment (jsteemann) [reply arangod/Cluster/SynchronizeShard.cpp:570]: Technically it is not necessary, and passing an empty std::string would have also worked.
I was using a pointer here intentionally to convey the idea that the value can be unset (i.e. nullptr).

**Review by dothebart — APPROVED** [review]

**Review by danielhlarkin — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]

**Pull request merged**